sim/midas/src/main/cc/simif.h

```diff
@@ -155,26 +172,72 @@ class simif_t {
 
   // End host-platform interface.
 
-  // LOADMEM functions
-  void read_mem(size_t addr, mpz_t &value);
-  void write_mem(size_t addr, mpz_t &value);
-  void write_mem_chunk(size_t addr, mpz_t &value, size_t bytes);
-  void zero_out_dram();
+  /// Returns the seed of the random-number generator.
+  uint64_t get_seed() const { return seed; };
 
-  uint64_t get_seed() { return seed; };
+  /// Returns the next available random number, modulo limit.
+  uint64_t rand_next(uint64_t limit) { return gen() % limit; }
 
-  // Returns the current target cycle of the fastest clock in the simulated
-  // system, based on the number of clock tokens enqueued (will report a larger
-  // number)
+  /**
+   * Returns the target cycle count.
+   *
+   * Provides the current target cycle of the fastest clock in the simulated
+   * system, based on the number of clock tokens enqueued (will report a larger
+   * number).
+   */
   uint64_t actual_tcycle();
-  // Returns the current host cycle as measured by a hardware counter
+
+  /// Returns the current host cycle as measured by a hardware counter
   uint64_t hcycle();
-  uint64_t rand_next(uint64_t limit) { return gen() % limit; }
 
+  /// Returns the last completed cycle number.
+  uint64_t get_end_tcycle() { return end_tcycle; }
+
+  // LOADMEM functions.
+  void read_mem(size_t addr, mpz_t &value);
+  void write_mem(size_t addr, mpz_t &value);
+  void write_mem_chunk(size_t addr, mpz_t &value, size_t bytes);
+
+private:
+  /// Waits for the target to be initialised.
```

Comment:
I don't like "target_init" it's more like simulator init -- "host_init" captured that better.

Reply:
I don't like this either, especially since it involves waiting for `INIT_DONE` and loading some memory contents. We should address it in the future, but for now I think `target_init` still captures the meaning since we are doing MMIO to the target and waiting for its response.